moto/sqs/models.py

```diff
@@ -478,6 +478,16 @@ def messages(self):
         ]
 
     def add_message(self, message):
+        if self.fifo_queue:
+            if self.attributes.get("ContentBasedDeduplication") == "true":
```

Comment:
Since the outer `if` has no `else`, you should be able to shorten this to:

```python
if self.fifo_queue and self.attributes.get("ContentBasedDeduplication") == "true":
```

Also, have you accounted for this case (from `aws sqs create-queue help`):

```
If  you  send one message with ContentBasedDeduplication enabled
and then another message with a MessageDeduplicationId  that  is
the  same  as  the one generated for the first MessageDeduplica-
tionId , the two messages are treated as duplicates and only one
copy of the message is delivered.
```

Reply:
Moved the condition to a single if and added the test case